src/ucs/sys/sys.c

```diff
@@ -1180,38 +1178,73 @@ void ucs_sys_cpuset_copy(ucs_cpu_set_t *dst, const ucs_sys_cpuset_t *src)
     }
 }
 
-ucs_sys_ns_t ucs_sys_get_ns(ucs_sys_ns_name_t name)
+ucs_sys_ns_t ucs_sys_get_ns(ucs_sys_namespace_type_t ns)
 {
-    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
     char filename[MAXPATHLEN];
     int res;
     struct stat st;
-    ucs_sys_ns_name_t ns;
 
-    if (name >= UCS_SYS_NS_LAST) {
+    if (ns >= UCS_SYS_NS_TYPE_LAST) {
         return 0;
     }
 
-    UCS_INIT_ONCE(&init_once) {
-        for (ns = UCS_SYS_NS_IPC; ns < UCS_SYS_NS_LAST; ns++) {
-            snprintf(filename, sizeof(filename), "%s/%s", UCS_PROCESS_NS_DIR,
-                     ucs_sys_namespace_info[ns].name);
+    snprintf(filename, sizeof(filename), "%s/%s", UCS_PROCESS_NS_DIR,
+             ucs_sys_namespace_info[ns].name);
 
-            res = stat(filename, &st);
-            if (res == 0) {
-                ucs_sys_namespace_info[ns].ino = (ucs_sys_ns_t)st.st_ino;
-            } else {
-                ucs_sys_namespace_info[ns].ino = ucs_sys_namespace_info[ns].dflt;
-            }
-        }
+    res = stat(filename, &st);
+    if (res == 0) {
+        return (ucs_sys_ns_t)st.st_ino;
     }
+    return ucs_sys_namespace_info[ns].dflt;
+}
 
-    return ucs_sys_namespace_info[name].ino;
+int ucs_sys_ns_is_default(ucs_sys_namespace_type_t ns)
+{
+    return ucs_sys_get_ns(ns) == ucs_sys_namespace_info[ns].dflt;
 }
 
-int ucs_sys_ns_is_root(ucs_sys_ns_name_t name)
+ucs_status_t ucs_sys_get_boot_id(uint64_t *high, uint64_t *low)
 {
-    ucs_sys_ns_t ns = ucs_sys_get_ns(name);
+    static ucs_init_once_t init_once = UCS_INIT_ONCE_INITIALIZER;
+    static uint64_t low_dword        = 0;
+    static uint64_t hi_dword         = 0;
+    static ucs_status_t status       = UCS_ERR_IO_ERROR;
+    const char *boot_id_filename     = "/proc/sys/kernel/random/boot_id";
+    const char *guid_scan_fmt        = "%x-%4hx-%4hx-%4hx-%2hhx%2hhx%2hhx%2hhx%2hhx%2hhx";
+    char guid_str[256];
+    ssize_t size;
+    int res;
+    uint32_t v1;
+    uint16_t v2;
+    uint16_t v3;
+    uint16_t v4;
+    uint8_t v5[6];
+
+    UCS_INIT_ONCE(&init_once) {
+        size = ucs_read_file_str(guid_str, sizeof(guid_str), 1,
+                                 "%s", boot_id_filename);
+        if (size <= 0) {
+            continue;
```

Comment:
what does that do?

Reply:
jump out of INIT_ONCE section, added comment

BTW, it seems there is error in INIT_ONCE comment: it is safe to use "continue", but "brake", direct jump out of section & return may lead to unexpected behavior

https://github.com/openucx/ucx/pull/4547